postings/roaring.go

```diff
@@ -68,21 +69,21 @@ func (d *roaringPostingsList) Intersect(other ImmutablePostingsList) error {
 	return nil
 }
 
-func (d *roaringPostingsList) Difference(other ImmutablePostingsList) error {
+func (d *roaringPostingsList) Difference(other List) error {
 	o, ok := other.(*roaringPostingsList)
 	if !ok {
 		return errDifferenceRoaringOnly
 	}
 
-	d.Lock()
 	o.RLock()
+	d.Lock()
```

Comment:
curious why you changed the order of the locks?

Reply:
This was just to be consistent with the other methods which acquire `o` first and then `d`.